lib/util.js

```diff
@@ -241,6 +242,41 @@ function inspectPromise(p) {
 
 
 function formatValue(ctx, value, recurseTimes) {
+
+  if (ctx.showProxy &&
+      (typeof value === 'object' || typeof value === 'function')) {
```

Comment:
You should check that `typeof value === 'object' && value !== null`, I think?

Style nit: can you drop the blank line?